src/decoder/plugins/VgmstreamDecoderPlugin.cxx

```diff
@@ -113,6 +115,59 @@ ParseContainerPath(Path path_fs)
 	return {path_fs.GetDirectoryName(), track};
 }
 
+static bool
+VgmstreamScanTagFile(Path path, TagHandler &handler, ReplayGainInfo *rgi) noexcept
+{
+	bool found_title = false;
+
+	const auto tags_path = AllocatedPath::Build(path.GetDirectoryName(), "!tags.m3u");
+	const std::string tags_path_utf8 = tags_path.ToUTF8();
+	libstreamfile_t *tags_file =
+		libstreamfile_open_from_stdio(tags_path_utf8.c_str());
+	if (tags_file == nullptr)
+		return false;
+
+	AtScopeExit(tags_file) { libstreamfile_close(tags_file); };
+
+	libvgmstream_tags_t *tags = libvgmstream_tags_init(tags_file);
+	if (tags == nullptr)
+		return false;
+
+	AtScopeExit(tags) { libvgmstream_tags_free(tags); };
+
+	libvgmstream_tags_find(tags, path.GetBase().c_str());
+
+	while (libvgmstream_tags_next_tag(tags)) {
+		handler.OnPair(tags->key, tags->val);
+
+		if (StringIsEqualIgnoreCase(tags->key, "TITLE"))
```

Comment:
Parsing this is redundant and adds unnecessary overhead. You already parse the key in that for loop below.